### DISCUSSION TITLE
Fully comply with the operation of the official website documents, cannot be used normally

### DISCUSSION THREAD

**Discussion by dickyangyi** [postalserver/postal#3000] [Help with using Postal] [answered]
![1](https://github.com/postalserver/postal/assets/28041890/48d9c8f4-b815-4148-8b5b-939848276b29)

https://docs.postalserver.io/getting-started

`yum install git curl jq
`
`git clone https://github.com/postalserver/install /opt/postal/install
sudo ln -s /opt/postal/install/bin/postal /usr/bin/postal
`
`docker run -d \
   --name postal-mariadb \
   -p 127.0.0.1:3306:3306 \
   --restart always \
   -e MARIADB_DATABASE=postal \
   -e MARIADB_ROOT_PASSWORD=postal \
   mariadb
`
`postal bootstrap postal.yourdomain.com
`
`postal initialize
postal make-user
`
`postal start
`
`docker run -d \
   --name postal-caddy \
   --restart always \
   --network host \
   -v /opt/postal/config/Caddyfile:/etc/caddy/Caddyfile \
   -v /opt/postal/caddy-data:/data \
   caddy
`
No matter in Centos 7 or Ubuntu server 22 installation, can not complete the installation, what is more terrible is that your documentation and videos are old versions, now is 3.3 version, I do not have a description or video for this version

**Comment by willpower232** [comment]
What problem did you encounter?

**Reply by dickyangyi** [reply]
> What problem did you encounter?

Installation is complete to open the https://postal.yourdomain.com, I completely is according to the official operation configuration, still can't use

**Reply by dickyangyi** [reply]
> What problem did you encounter?

I checked carefully. I didn't miss a step

**Reply by dickyangyi** [reply]
postal.yml
```dockerfile
version: 2

postal:
  web_hostname: mail.zebao.co
  web_protocol: https
  smtp_hostname: mail.zebao.co

main_db:
  host: 127.0.0.1
  username: root
  password: postal
  database: postal

message_db:
  host: 127.0.0.1
  username: root
  password: postal
  prefix: postal

smtp_server:
  default_bind_address: "::"

dns:
  # Specify the DNS records that you have configured. Refer to the documentation at
  # https://github.com/atech/postal/wiki/Domains-&-DNS-Configuration for further
  # information about these.
  mx_records:
    - mx.mail.zebao.co
  spf_include: spf.mail.zebao.co
  return_path_domain: rp.mail.zebao.co
  route_domain: routes.mail.zebao.co
  track_domain: track.mail.zebao.co

smtp:
  # Specify an SMTP server that can be used to send messages from the Postal management
  # system to users. You can configure this to use a Postal mail server once the
  # your installation has been set up.
  host: 127.0.0.1
  port: 2525
  username: # Complete when Postal is running and you can
  password: # generate the credentials within the interface.
  from_name: Postal
  from_address: mail.zebao.co

rails:
  # This is generated automatically by the config initialization. It should be a random
  # string unique to your installation.
  secret_key: 56ed7252f58708d365e09d5849db2095a6da0f831af37ddcaeb769a98e14d171b28c2b8a28a3a72417ba7b3e970c2892606810d7c7c7993d68901ff248a19e0dddce06ddbd32a4c1896df9300817347363f345bfd961519cef106f1185ef809e902fa3ca7fe082063d6184a76078549c9e820696fad06692f51ead8d19755994

```

**Reply by dickyangyi** [reply]
Caddyfile

```
# Update this to match the hostname you wish to use for accessing the
# Postal web interface.
mail.zebao.co {
  reverse_proxy 127.0.0.1:5000
}

# If you use open/click tracking you may need to add additional hosts
# to this file using the format below.
#
# click.yourdomain.com {
#   reverse_proxy 127.0.0.1:5000 {
#     header_up X-Postal-Track-Host "1"
#   }
# }
```

**Reply by willpower232** [reply, marked as answer]
it looks like you haven't pointed mail.zebao.co to your servers public IP address

**Reply by dickyangyi** [reply]
> it looks like you haven't pointed mail.zebao.co to your servers public IP address

Oh my God, this is my local deployment of the test server, not deployed to the cloud before finding out, which needs to be deployed in the cloud and comply with the DNS configuration to access the web?

**Reply by willpower232** [reply]
You need a domain name which points at your deployment in order for caddy to work.

I don't personally know enough about caddy to turn off the https part but presumably you could turn it off and use a different domain name so it doesn't have to publicly create an SSL certificate.

Otherwise you could use something else like nginx which does not try to create a certificate for you.

**Reply by dickyangyi** [reply]
> You need a domain name which points at your deployment in order for caddy to work.
> 
> I don't personally know enough about caddy to turn off the https part but presumably you could turn it off and use a different domain name so it doesn't have to publicly create an SSL certificate.
> 
> Otherwise you could use something else like nginx which does not try to create a certificate for you.

Ok, this is my local deployment. After caddy is installed, I can get through port 80, but it still cannot be used

**Reply by willpower232** [reply]
What error message do you see on your browser? That would be the best clue as to what is happening.

**Reply by dickyangyi** [reply]
> What error message do you see on your browser? That would be the best clue as to what is happening.

Sorry for the late reply. Now I have moved to AWS for use. The local environment failed due to DNS problems, thank you!
